### ISSUE TITLE
Adding black command to docker-travis script to run automatically on each commit

### ISSUE THREAD

**Issue by mtreacy002** [anitab-org/mentorship-backend#483]
## Description
As a developer, I would like it if code can be refactored to PEP8 standard automatically every time I commit.
I need to add black to the script within docker/travis,
so that I can ensure code quality of my code changes.
Note: Good reference to start with: https://docs.travis-ci.com/user/docker/#pushing-a-docker-image-to-a-registry

## Mocks)

If we can do this, we can prevent issues in code quality creeping up in our development.

- before adopting black:

<img width="807" alt="Screen Shot 2020-02-29 at 1 16 45 am" src="https://user-images.githubusercontent.com/29667122/76152135-181ac500-6110-11ea-9888-949bac3ab12d.png">

- after adopting black:

<img width="814" alt="Screen Shot 2020-02-29 at 1 06 26 am" src="https://user-images.githubusercontent.com/29667122/76152125-00dbd780-6110-11ea-9f68-7166d48816a4.png">

## Acceptance Criteria
Should not break the system

### Update [Required]
- [ ] Not needed. Black already in pip dependencies
### Enhancement to Update [Optional]
- [ ] Not sure

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 day

**Comment by isabelcosta** [comment]
interesting point by @ramitsawhney27 👉 [Zulip message](https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/Code.20Quality/near/190438725)

**Comment by bera5186** [comment]
@isabelcosta I would like to work upon this issue

**Comment by isabelcosta** [comment]
@bera5186 I assigned you an issue already so I won't assign you this one.

**Comment by bera5186** [comment]
@isabelcosta I have sent a [Pull request](https://github.com/anitab-org/mentorship-backend/pull/500) to previously assigned [Issue](https://github.com/anitab-org/mentorship-backend/issues/493) to me. Till you review my pull request I would love to work upon this issue. Can you assign me this.

**Comment by mtreacy002** [comment]
@bera5186, have you started working on this issue? If you haven't, then I would like to give it a try (I have a bit more time now since I'm almost done with my proposal - I hope 🤞- just waiting for mentors/admins feedback). Please let me know. 😉

**Comment by isabelcosta** [comment]
I will assign this to you as mentioned on the Ask Me Anything session @mtreacy002

**Comment by mtreacy002** [comment]
Thanks, @isabelcosta . Actually I have just pushed a PR #511 for this. can you pls check?

**Comment by isabelcosta** [comment]
Later in the day I can

**Comment by mtreacy002** [comment]
The thing is, for it to work on your Travis, you need to set the Travis environment for the DOCKER_USERNAME and DOCKER_PASSWORD

**Comment by mtreacy002** [comment]
Update.
@ramitsawhney27 and @isabelcosta.

I have to close this issue as I found out when I worked through PR #511, Black actually removing few important things we want to be in our codes. Here are the examples of what it did:

- it removed the parameters specification variables which is allowed by [PEP612](https://www.python.org/dev/peps/pep-0612/) and the type hint for TypedDict which is allowed in [PEP589](https://www.python.org/dev/peps/pep-0589/)
<img width="504" alt="Screen Shot 2020-03-31 at 8 22 59 pm" src="https://user-images.githubusercontent.com/29667122/78085876-ff988480-7407-11ea-91c1-243c6850e25e.png">

- this one is even worse. It randomly deleted a section in .md that we intentionally put there.... 😤 😡 .

<img width="1664" alt="Screen Shot 2020-03-31 at 9 08 03 pm" src="https://user-images.githubusercontent.com/29667122/78085967-3bcbe500-7408-11ea-8af4-4b8675be0894.png">

So, I'm not impressed. Until I can find a way to mute some of it's aggressive behaviour (like using pep8 --ignore), I will put aside this idea of relying on Black to reformat our code. We can still use it manually for code maintenance every few commits, that way we can select which changes to take from its recommendation.

But I will open another issue to keep the idea of building docker image in travis ci which should help us with app scalability in case we want to scale it horizontally and spread our images across different servers on different network zones. 😉
